### PR TITLE
[LLVM Analyzer][FastSim] Cleanup dead assignments reported by llvm analyzer

### PR DESCRIPTION
**Pull request by iarspider**
#### PR description:

This fixes clang-analyzer warnings https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_14_2_X_2024-09-29-0000/el8_amd64_gcc12/llvm-analysis/index.html

#### PR validation:

Bot tests

### CODE REVIEW

File: FastSimulation/CaloGeometryTools/src/CrystalWindowMap.cc [hunk 1]
```diff
@@ -52,7 +52,7 @@ bool CrystalWindowMap::getCrystalWindow(unsigned iq, std::vector<unsigned>& cw)
 bool CrystalWindowMap::getCrystalWindow(unsigned iq, const std::vector<unsigned>* cw) const {
   if (iq < size_)  // iq >= 0, since iq is unsigned
   {
-    cw = &myNeighbours_[iq];
+    [[clang::suppress]] cw = &myNeighbours_[iq];
```
Comment (iarspider) [inline FastSimulation/CaloGeometryTools/src/CrystalWindowMap.cc:55]: No idea why clang reports dead assignment here

File: FastSimulation/CTPPSFastGeometry/src/CTPPSToFDetector.cc [hunk 2]
```diff
@@ -81,27 +82,21 @@ void CTPPSToFDetector::AddHit(double x, double y, double tof) {
...
-  // first, get the row number
-  unsigned int i;
-  unsigned int start_idx = 0;
-  unsigned int end_idx = cellRow_.size();
-  for (i = 0; i < cellRow_.size(); i++) {
-    if (y >= cellRow_.at(i).first && y <= cellRow_.at(i).second)
-      break;
-  }
-  if (i >= cellRow_.size())
+  auto it = std::find_if(
```
Comment (iarspider) [inline FastSimulation/CTPPSFastGeometry/src/CTPPSToFDetector.cc:85]: Values of start_idx and end_idx were not used ([overwritten later](https://github.com/cms-sw/cmssw/pull/46255/files#diff-17d81d12169ae03f5dcddf62becdc9b805803687af260d7eca6d6d4464475ce0L96-L97)).

File: FastSimulation/CTPPSFastGeometry/src/CTPPSToFDetector.cc [hunk 3]
```diff
@@ -81,27 +82,21 @@ void CTPPSToFDetector::AddHit(double x, double y, double tof) {
...
+      cellRow_.begin(), cellRow_.end(), [y](const auto& cell) { return y >= cell.first && y <= cell.second; });
+
+  if (it == cellRow_.end())
     return 0;
-  y_idx = i + 1;
-  start_idx = 0;
-  end_idx = cellColumn_.size();
-  for (i = start_idx; i < end_idx; i++) {
-    if (x <= cellColumn_.at(i).first && x > cellColumn_.at(i).second)
```
Comment (iarspider) [inline FastSimulation/CTPPSFastGeometry/src/CTPPSToFDetector.cc:99]: this implies that `cellColumn_[i].first >= cellColumn_[i].second` - is that true?

**Comment by iarspider** [comment]
please test

File: FastSimulation/CaloGeometryTools/src/CrystalWindowMap.cc (hunk 1 above)
Comment (civanch) [reply FastSimulation/CaloGeometryTools/src/CrystalWindowMap.cc:55]: @iarspider , @kpedro88 , I would use parentecies in this assignment.

File: FastSimulation/CTPPSFastGeometry/src/CTPPSToFDetector.cc (hunk 2 above)
Comment (civanch) [reply FastSimulation/CTPPSFastGeometry/src/CTPPSToFDetector.cc:85]: The algorithm should be understood in general.

**Comment by iarspider** [comment]
please test

**Comment by civanch** [comment]
@kpedro88 , @fabferro , clang analyzer identify problems in few CTPPS FastSim classes. Most of proposed fixes are trivial and can go without any problem but proposed modifications in CTPPSToFDetector.cc are substantial. Does this code used by any PPS application? Or it may be obsolete?

**Comment by AndreaBellora** [comment]
> My understanding is that CTPPS inside FastSim is an obsolete project overcome by DirectSimulation. @diemort and @AndreaBellora can you confirm?

I tend to agree, but, to be fair, I've never heard of a CTPPS FastSim project. It might as well have been a thing before I joined the group, though. Unless @diemort remembers otherwise, I would GL the changes.

**Comment by civanch** [comment]
+1

Thanks PPS colleagues. This PR may safely be merged. If not needed, the project may be removed at any time.

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**